specs/phase1/data-availability-sampling.md

```diff
@@ -0,0 +1,218 @@
+# Ethereum 2.0 Phase 1 -- Data Availability Sampling
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Custom types](#custom-types)
+- [Configuration](#configuration)
+  - [Misc](#misc)
+- [New containers](#new-containers)
+  - [DASSample](#dassample)
+  - [ShardBlob](#shardblob)
+  - [SignedShardBlob](#signedshardblob)
+- [Helper functions](#helper-functions)
+  - [Reverse bit ordering](#reverse-bit-ordering)
+    - [`reverse_bit_order`](#reverse_bit_order)
+    - [`reverse_bit_order_list`](#reverse_bit_order_list)
+  - [Data extension](#data-extension)
+  - [Data recovery](#data-recovery)
+- [DAS functions](#das-functions)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Custom types
+
+We define the following Python custom types for type hinting and readability:
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SampleIndex` | `uint64` | A sample index, corresponding to chunk of extended data |
+| `BLSPoint` | `uint256` | A number `x` in the range `0 <= x < MODULUS` |
+
+
+## Configuration
+
+### Misc
+
+| Name | Value | Notes |
+| - | - | - |
+| `MAX_RESAMPLE_TIME` | `TODO` (= TODO) | Time window to sample a shard blob and put it on vertical subnets |
+
+
+## New containers
+
+### DASSample
+
+```python
+class DASSample(Container):
+    slot: Slot
+    shard: Shard
+    index: SampleIndex
+    proof: BLSCommitment
+    data: Vector[BLSPoint, POINTS_PER_SAMPLE]
```

Comment:
`POINTS_PER_SAMPLE` is undefined

Reply:
It's in an odd place in the base branch, in the phase 1 beacon-chain doc. I left it there to avoid conflicts. It's not missing.